lib/zif/compound_sprite.rb

```diff
@@ -103,14 +103,13 @@ def source_rect
     # DRGTK docs on #draw_override:
     # http://docs.dragonruby.org/#----performance---static-sprites-as-classes-with-custom-drawing---main.rb
     # @api private
-    def draw_override(ffi_draw)
+    def draw_override(ffi_draw, parent_transform={offset: {x: 0, y: 0}, zoom: {x: 1.0, y: 1.0}})
       # $services&.named(:tracer)&.mark("CompoundSprite(#{@name})#draw_override: begin")
       # Treat an alpha setting of 0 as an indication that it should be hidden, to match Sprite behavior
       return if @a.zero?
 
-      view_actual_size! unless source_is_set?
+      trans = transform(parent_transform)
```

Comment:
I am a little concerned about the allocation of objects here, because this code path can be the performance bottleneck for games using large numbers of sprites in activelayers.  Since this is a private API I'd like to use a little more complex pattern of individual params instead of allocating hashes. `trans_offset_x: 0, trans_offset_y: 0, trans_zoom_x: 1.0, trans_zoom_y: 1.0`

Then we should calculate the transformation of each individually; instead of in a single method returning an object we might need 4 methods returning floats.

This removes the need to perform `{}.merge()` in the calculation as well.